packages/common/src/search/_internal/hubSearchEventAttendees.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  GetRegistrationsParams,
+  IGetRegistrationsParams,
+} from "../../events/api";
+import {
+  getRegistrations,
+  IPagedRegistrationResponse,
+} from "../../events/api/orval/api/orval-events";
+import { eventAttendeeToSearchResult } from "../../events/api/utils/search";
+import { IPredicate, IQuery } from "../types/IHubCatalog";
+import { IHubSearchOptions } from "../types/IHubSearchOptions";
+import { IHubSearchResponse } from "../types/IHubSearchResponse";
+import { IHubSearchResult } from "../types/IHubSearchResult";
+
+export const processGetRegistrationsParams = (
+  options: IHubSearchOptions,
+  query: IQuery
+): GetRegistrationsParams => {
+  // valid eys to translate from IQuery predicates
+  const validPredicateKeys: Array<keyof IPredicate> = [
+    "eventId",
+    "userId",
+    "role",
+    "status",
```

Comment:
The [figma](https://www.figma.com/file/WiYh5XAkNu3uk8Kh4P3I4T/Events-3.0---First-release-to-unblock-workspaces?type=design&node-id=45-243753&mode=design&t=MVMP7SxW4EZBejht-0) suggests we should be able to filter attendees by free-form text input that matches against `first`, `last`, and `username`. The API doesn't support this _yet_, I asked Jeremy to add another enhancement for this, until then GetRegistrationsParams will not have a property for this input. That value will come in as a `term` predicate. I'll update my IQuery examples from earlier comments to show this. We should still be able to hook this up to the expected property in the `data` object we pass to `getRegistrations` once Jeremy adds the ticket, assuming we know what the name of the property will be. Typescript might complain b/c that prop doesn't yet exist on the GetRegistrationsParams interface, but we should be able to add a `// @ts-ignore` comment to suppress that error. The API should ignore the value passed on the XHR until the back-end work is completed/released.